src/Symfony/Component/EventDispatcher/InvokableEventSubscriberTrait.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\EventDispatcher;
+
+/**
+ * Helper trait for single-event subscription.
+ *
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+trait InvokableEventSubscriberTrait
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [self::getEventName() => ['__invoke', self::getPriority()]];
+    }
+
+    private static function getEventName(): string
+    {
+        $method = new \ReflectionMethod(self::class, '__invoke');
+
+        if (0 === $method->getNumberOfParameters()) {
+            throw new \LogicException(sprintf('The Event name cannot be inferred from "%s::%s()" method without parameters, you must to define one or implement the "%s::getEventName()" method to return it explicitly.', self::class, $method->getName(), self::class));
+        }
+
+        $parameter = $method->getParameters()[0];
+
+        if (null === $type = $parameter->getType()) {
+            throw new \LogicException(sprintf('The Event name cannot be inferred from "%s::%s()" method without type-hint the "$%s" parameter, you must to type-hint it or implement the "%s::getEventName()" method to return it explicitly.', self::class, $method->getName(), $parameter->getName(), self::class));
+        }
+
+        if ($type->isBuiltin()) {
+            throw new \LogicException(sprintf('The "$%s" parameter defined in "%s::%s()" method must be a class, "%s" given.', $parameter->getName(), self::class, $method->getName(), $type->getName()));
```

Comment:
What is the use case for overriding `getEventName` name instead of type-hinting?

Reply:
* You can listen to events that are not dispatched/aliased as FQCN.
* If you don't need the event object, you don't have to type-hint an unused parameter.

Reply:
Same opinion: both use cases are solved by implementing EventSubscriberInterface to me and we should not provide alternatives within this trait. Otherwise, we're going to have to teach ppl that this extension point exists and when to use it, while it would be better focus on teaching them the only useful stuff on the long run: EventSubscriberInterface

Reply:
I agree about making them implement EventSubscriberInterface itself for such case rather than using the trait.
We could even drop defining `getEventName` entirely, reducing our BC surface for the trait (as everything in a public trait is subject to BC, including private members).